#151948278 Front tests

machariamarigi opened the pull request:
### What does this PR do?

- Add tests to the front application

### Description of Task to be completed?

- Go through each test file and follow the instructions to fix the tests. The instructions are distributed throughout each file and the instruction lines begin with ###

### How should this be manually tested?

- You can test this manually by issuing this command; `./manage.py test hc/front` from the project root directory

### What are the relevant pivotal tracker stories?

- See [ Chore #151948278](https://www.pivotaltracker.com/story/show/151948278)

#### Screenshots (if appropriate)
#### Questions:

Sekams committed:
Add gunicorn

Add gunicorn dependency to the requirements file

Sekams committed:
Add gunicorn

Add gunicorn dependency to the requirements file

Sekams committed:
Add Procfile for Heroku deployment

Add Procfile containing the configuration for successful deployment to Heroku

Sekams committed:
Configure Heroku database

Add code for configuring the Heroku database from the environment variables

Sekams committed:
Configure Heroku database

Add code for configuring the Heroku database from the environment variables

Sekams committed:
Add dj_database_url package

Add dj_database_url package to the requirements file

Sekams committed:
Add dj_database_url package

Add dj_database_url package to the requirements file

Sekams committed:
Add whitenoise package

Add whitenoise package to the application for accessing static files

Sekams committed:
Add whitenoise package

Add whitenoise package to the application for accessing static files

Sekams committed:
Add whitenoise package

Add whitenoise package to the application for accessing static files

Sekams committed:
Add whitenoise package

Add whitenoise package to the application for accessing static files

Sekams committed:
Add whitenoise package

Add whitenoise package to the application for accessing static files

Sekams committed:
Add whitenoise package

Add whitenoise package to the application for accessing static files

Sekams committed:
Add DB variable for Travis

Add the variable DB for successful travis builds

Sekams committed:
Add DB variable for Travis

Add the variable DB for successful travis builds

Sekams committed:
Add DB variable for Travis

Add the variable DB for successful travis builds

Sekams committed:
Add DB variable for Travis

Add the variable DB for successful travis builds

Sekams committed:
fix(settings): Separate Heroku Configuration

Make separate configuration for Heroku deployment

Sekams committed:
fix(settings): Separate Heroku Configuration

Make separate configuration for Heroku deployment

Sekams committed:
fix(settings): Separate Heroku Configuration

Make separate configuration for Heroku deployment

Sekams committed:
fix(settings): Separate Heroku Configuration

Make separate configuration for Heroku deployment

Sekams committed:
fix(settings): Separate Heroku Configuration

Make separate configuration for Heroku deployment

Sekams committed:
fix(settings): Separate Heroku Configuration

Make separate configuration for Heroku deployment

Commit:
Ignore .env and vscode editor configurations

Commit:
Ignore .env and vscode editor configurations

Commit:
Ignore .env and vscode editor configurations

Commit:
Ignore .env and vscode editor configurations

Commit:
Ignore .env and vscode editor configurations

Commit:
Ignore .env and vscode editor configurations

Commit:
Install mock library

Commit:
Install mock library

Commit:
Install mock library

Commit:
Install mock library

Commit:
Remove unnecessary libraries

Commit:
Write tests for team access and bad_kinds

Commit:
Write tests for team access and bad_kinds

Commit:
Write tests for team access for checks

Commit:
Write tests for team access for checks

Commit:
Write tests for validating priorities in pushovers

Commit:
Write tests for validating priorities in pushovers

Commit:
Add whitenoise library

Commit:
Add whitenoise library

Commit:
Add whitenoise library

Commit:
Add whitenoise library

Commit:
Remove deployment code

Commit:
Remove deployment code

Commit:
Remove deployment code

Commit:
Remove deployment code

Commit:
Remove deployment code

Commit:
Remove deployment code

Commit:
Remove whitenoise middleware

Commit:
Remove whitenoise middleware

Commit:
Configure the app to use smtp hosts

Commit:
Configure the app to use smtp hosts

ivantoz commented on hc/front/tests/test_add_channel.py:
```diff
@@ -37,5 +37,24 @@ def test_instructions_work(self):
             r = self.client.get(url)
             self.assertContains(r, "Integration Settings", status_code=200)
 
-    ### Test that the team access works
```
Any reasons these tests comments were removed

ivantoz commented on hc/front/tests/test_add_channel.py:
```diff
@@ -37,5 +37,24 @@ def test_instructions_work(self):
             self.assertContains(r, "Integration Settings", status_code=200)
 
-    ### Test that the team access works
-    ### Test that bad kinds don't work
```
Also this

ivantoz commented on hc/front/tests/test_add_check.py:
```diff
@@ -11,4 +11,12 @@ def test_it_works(self):
         self.assertRedirects(r, "/checks/")
         assert Check.objects.count() == 1
 
-    ### Test that team access works
```
And this

ivantoz commented on hc/front/tests/test_add_pushbover.py:
```diff
@@ -37,4 +37,13 @@ def test_it_validates_nonce(self):
         r = self.client.get("/integrations/add_pushover/?%s" % params)
         assert r.status_code == 403
 
-    ### Test that pushover validates priority
```
this too

ivantoz commented on hc/settings.py:
```diff
@@ -88,6 +88,8 @@
     }
 }
 
+
```
Any reason for introduction of these empty lines

Commit:
[CHORE]Get email configurations from the enviroment

Commit:
[CHORE]Get email configurations from the enviroment

Arthur236 committed:
[Chore #151948277] Add test for login and redirect already logged in

Arthur236 committed:
[Chore #151948277] Add test for login and redirect already logged in

Arthur236 committed:
[Chore #151948277] Add test for login with bad token and redirects

Arthur236 committed:
[Chore #151948277] Add test for login with bad token and redirects

Arthur236 committed:
[Chore #151948277] Add test to assert that a user was created

Arthur236 committed:
[Chore #151948277] Add test to assert that a user was created

Arthur236 committed:
[Chore #151948277] Add test to assert email contents

Arthur236 committed:
[Chore #151948277] Add test to assert email contents

Arthur236 committed:
[Chore #151948277] Add test to assert that check is associated with the new user

Arthur236 committed:
[Chore #151948277] Add test to assert that check is associated with the new user

Arthur236 committed:
[Chore #151948277] Add test to assert that token is set

Arthur236 committed:
[Chore #151948277] Add test to assert that token is set

Arthur236 committed:
[Chore #151948277] Add test to assert that email was sent and check email content

Arthur236 committed:
[Chore #151948277] Add test to assert that email was sent and check email content

Arthur236 committed:
[Chore #151948277] Add test to assert that email was sent and check email content

Arthur236 committed:
[Chore #151948277] Add test to assert that email was sent and check email content

Arthur236 committed:
[Chore #151948277] Add test to assert the existence of member emails

Arthur236 committed:
[Chore #151948277] Add test to assert the existence of member emails

Arthur236 committed:
[Chore #151948277] Add test to assert that email was sent and check email content

Arthur236 committed:
[Chore #151948277] Add test to assert that email was sent and check email content

Arthur236 committed:
[Chore #151948277] Add test to assert that it creates and revokes api key

Arthur236 committed:
[Chore #151948277] Add test to assert that it creates and revokes api key

Arthur236 committed:
[Chore #151948277] Add test to assert that it redirects

Arthur236 committed:
[Chore #151948277] Add test to assert that it redirects

Arthur236 committed:
[Chore #151948277] Add test to assert that it returns correct status codes

Arthur236 committed:
[Chore #151948277] Add test to assert that it returns correct status codes

Arthur236 committed:
[Chore #151948277] Add test to assert that the profile object exists and it belongs to the user

Arthur236 committed:
[Chore #151948277] Add test to assert that the profile object exists and it belongs to the user

Commit:
[CHORE]Add heroku site root

Commit:
[CHORE]Add heroku site root

Commit:
[CHORE]Add heroku site root

Commit:
[CHORE]Add heroku site root

Commit:
[CHORE]Add heroku site root

Commit:
[CHORE]Add heroku site root

Commit:
[CHORE]Add heroku site root

Sekams committed:
Merge branch 'develop' into ch-front-tests-151948278

Sekams committed:
refactor(conflicts): Refactor code to resolve conflicts

- Refactor code in gitignore and requirements files to resolve conflicts with the develop branch

[Fixes #151948278]

Sekams committed:
refactor(conflicts): Refactor code to resolve conflicts

- Refactor code in gitignore and requirements files to resolve conflicts with the develop branch

[Fixes #151948278]

Sekams committed:
refactor(conflicts): Refactor code to resolve conflicts

- Refactor code in gitignore and requirements files to resolve conflicts with the develop branch

[Fixes #151948278]

Sekams committed:
refactor(conflicts): Refactor code to resolve conflicts

- Refactor code in gitignore and requirements files to resolve conflicts with the develop branch

[Fixes #151948278]

Arthur236 committed:
[Chore #151948277] Change assert comment to read empty string instead instead of none

Arthur236 committed:
[Chore #151948277] Change assert comment to read empty string instead instead of none

Arthur236 committed:
[Chore #151948277] Change assert method for checking if token is not none

Arthur236 committed:
[Chore #151948277] Change assert method for checking if token is not none

Commit:
[CHORE]Get SITE_ROOT configuration from the environment

Commit:
[CHORE]Get SITE_ROOT configuration from the environment

Commit:
[CHORE]Get SITE_ROOT configuration from the environment

Commit:
[CHORE]Get SITE_ROOT configuration from the environment

Commit:
[CHORE]Get SITE_ROOT configuration from the environment

Commit:
[CHORE]Get SITE_ROOT configuration from the environment

Commit:
[CHORE]Get SITE_ROOT configuration from the environment

Commit:
[Chore #152238709]Update README.md

Commit:
[Chore #152238709]Update README.md

Commit:
[Chore #152238709]Resolve conflicts

Commit:
[Chore #152238709]Resolve conflicts

Commit:
[Chore #152238709]Resolve conflicts

Commit:
[Chore #152238709]Resolve conflicts

Commit:
[Chore #151948278] Ignore .env file

Commit:
[Chore #151948278] Ignore .env file

Commit:
[Chore #151948278] Ignore .env file

Commit:
[Chore #151948278] Add comments describing tests added
